packages/core/src/events.rs

```diff
@@ -654,7 +654,26 @@ impl<T> ListenerCallback<T> {
     pub fn call(&self, event: Event<dyn Any>) {
         let runtime = Runtime::current().expect("ListenerCallback must be called within a runtime");
         runtime.with_scope_on_stack(self.origin, || {
-            (self.callback.borrow_mut())(event);
+            // Use try_borrow_mut to handle cases where the callback is already borrowed
+            // This provides additional safety for edge cases where events might be nested
+            match self.callback.try_borrow_mut() {
+                Ok(mut callback) => {
+                    callback(event);
+                }
+                Err(_) => {
+                    // If we can't borrow mutably, it means we're in a reentrant call
+                    // This should be very rare now that we've fixed the main borrow issue,
+                    // but we defer the event to be safe
+                    let callback = self.callback.clone();
+                    let event = event.clone();
+
+                    crate::spawn(async move {
+                        if let Ok(mut cb) = callback.try_borrow_mut() {
+                            cb(event);
+                        }
+                    });
+                }
+            }
```

Comment:
Haven't looked through the whole PR, but the BorrowMut error happens only in WASM after the app has panicked once. The runtime refcell is left in an open state. Since panics don't unwind in WASM, the Ref is not dropped and the refcell is stuck open.

When the next event is handled on the javascript side, we call a WASM callback regardless even if the Rust side has crashed.

I'd like to fix this in WASM, but we have to work around the fact that there aren't unwind tables and we won't know how to bring the app back into a recovered state.